cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -336,8 +337,29 @@ XDMFFile::read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
 
   const auto [entities, eshape] = read_topology_data(name, xpath);
 
-  pugi::xml_node values_data_node
-      = grid_node.child("Attribute").child("DataItem");
+  pugi::xml_node attribute_node = grid_node.child("Attribute");
+  pugi::xml_node values_data_node = attribute_node.child("DataItem");
+  if (not attribute_label.empty())
+  {
+    bool found = false;
+    for (; attribute_node;
+         attribute_node = attribute_node.next_sibling("Attribute"))
+    {
+      pugi::xml_attribute hint;
+      pugi::xml_attribute name = attribute_node.attribute("Name", hint);
+      if (std::string(name.value()) == attribute_label)
```

Comment:
Creating the string is a bit clumsy, maybe use https://en.cppreference.com/w/cpp/string/char_traits/compare.

Reply:
Using the function you suggested requires providing a number of characters to compare, so it would have to be something like

```c++
if (auto len = std::max(std::strlen(name.value()), attribute_label.size());
    std::char_traits<char>::compare(name.value(), attribute_label.c_str(),
                                    len) == 0)
```
right?

Doesn't look less clumsy to me.

Reply:
Done!